File: static/CWE-79/javascript/vulnerable.js
// CWE-79: Improper Neutralization of Input During Web Page Generation ('Cross-site Scripting')
// Untrusted request input is reflected into an HTML response without escaping.
// Vulnerable sink: res.send(`<h1>Hello ${req.query.name}</h1>`)

const express = require('express');

const app = express();

app.get('/greet', (req, res) => {
  const name = req.query.name;
  res.send(`<html><body><h1>Hello ${name}</h1></body></html>`);
});

app.listen(3000);

File: static/CWE-79/java/vulnerable.java
// CWE-79: Improper Neutralization of Input During Web Page Generation (XSS)
// User-supplied request parameter is written into the HTTP response without encoding.
// Vulnerable sink: HttpServletResponse.getWriter().println(request.getParameter(...))

import java.io.PrintWriter;
import javax.servlet.http.HttpServletRequest;
import javax.servlet.http.HttpServletResponse;

class CWE_79 {
    public void handle(HttpServletRequest request, HttpServletResponse response) throws Exception {
        PrintWriter out = response.getWriter();
        out.println("<h1>Hello " + request.getParameter("name") + "</h1>");
    }
}
